src/lib/I18n/index.js

```diff
@@ -79,9 +93,10 @@ export default class I18n {
   get currentLocale() {
     return RUNTIME.locale;
   }
+
   // eslint-disable-next-line class-methods-use-this
-  get setLocale() {
-    return setLocale;
+  setLocale(...args) {
+    return this.pInitiate.then(() => setLocale(...args));
```

Comment:
Please make this into an async function and use await. Also check for this.initiated to skip the wait.